mgplvm/training.py

```diff
@@ -188,10 +188,21 @@ def svgp(Y,
          ts=None,
         batch_pool = None,
         hook = None,
-        neuron_idxs = None):
+        neuron_idxs = None,
+        loss_margin=0.,
+        stop_iters=10):
     '''
+    max_steps [optional]: int, default=1000
+        maximum number of training iterations
+    
     batch_pool [optional] : None or int list
         pool of indices from which to batch (used to train a partial model)
+        
+    loss_margin [optional] : float, default=0
```

Comment:
I think I agree that it would be good to have a user-defined stopping condition with 1-2 sensible default options.
One thing we need to be careful with re. flexible stopping is that it might not work well with batching unless we make sure to average the loss across multiple batches. Otherwise having a high-likelihood batch will cause stopping even early in training if you have a lot of data.

Reply:
Yes I like the idea of user-defined conditions, looks good thanks! The reason the margin could be positive is I observed sometimes the loss would go up for a bit before going down more, may be an indication the learning rate was not well chosen though.

Reply:
Okay great I'll try and implement these changes in this branch